openedx/adg/lms/applications/models.py

```diff
@@ -341,14 +342,33 @@ def get_user_enrolled_course(self, user):
         Returns:
             CourseOverview: Enrolled multilingual course or None
         """
-        enrolled_multilingual_course = self.multilingual_courses(
-            manager='open_multilingual_courses'
-        ).filter(
+        if user.is_anonymous:
+            return None
+
+        enrolled_multilingual_course = self.open_multilingual_courses.filter(
             course__courseenrollment__user=user,
             course__courseenrollment__is_active=True
         ).first()
         return enrolled_multilingual_course.course if enrolled_multilingual_course else None
-    # pylint: enable=no-member
+
+    def get_preferred_lang_course(self):
+        """
+        Return course with preferred language.
+
+        If course with preferred lang is not found then return the first course from the group.
+        If there are no open courses then return None.
+
+        Returns:
+            MultilingualCourse: User preferred lang course.
+        """
+        user_preferred_lang = get_language()
+        open_multilingual_courses = self.open_multilingual_courses
+
+        for multilingual_course in open_multilingual_courses:
+            if multilingual_course.course.language == user_preferred_lang:
+                return multilingual_course.course
+
+        return open_multilingual_courses[0].course if open_multilingual_courses else None
```

Comment:
Let's please follow single responsibility principle, Return None it preffered lang course is not found.